http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/GceRuntimeMonitor.scala

```diff
@@ -0,0 +1,54 @@
+package org.broadinstitute.dsde.workbench.leonardo.monitor
+
+import java.time.Instant
+
+import cats.mtl.ApplicativeAsk
+import fs2.Stream
+import org.broadinstitute.dsde.workbench.google2.ZoneName
+import org.broadinstitute.dsde.workbench.leonardo._
+import org.broadinstitute.dsde.workbench.leonardo.config.{ImageConfig, RuntimeBucketConfig}
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+
+import scala.concurrent.duration._
+
+/**
+  * Monitor for runtime status transition: Starting, Creating, Deleting, Stopping
+  * It doesn't trigger any of the action but only responsible for monitoring the progress and make necessary cleanup when the transition is done
+  */
+trait GceRuntimeMonitor[F[_]] {
```

Comment:
Haven't reviewed it all yet, but I'm wondering how much of this is GCE-specific. What would it take to abstract this monitoring approach to Dataproc and GKE?

Should we have something similar to `RuntimeAlgebra[F[_]]` which has interpreters for GCE and Dataproc?

Reply:
yea...I think eventually we'll have something like `MonitorAlgebra`...but each implementation is completely cloud service specific